src/Microsoft.Health.Dicom.Functions.Client/DicomAzureFunctionsClient.cs

```diff
@@ -195,4 +215,27 @@ private static IOperationCheckpoint ParseCheckpoint(DicomOperation type, Durable
                 throw new ArgumentOutOfRangeException(nameof(type));
         }
     }
+
+    private async Task<DurableOrchestrationStatus> GetStatusInternalAsync(Guid operationId)
```

Comment:
Perhaps we can include a bit more in this method, like:
```csharp
private async Task<T> GetStatusAsync(Guid operationId, Func<DicomOperation, DurableOrchestrationStatus, IOperationCheckpoint, Task<T>> factoryAsync)
    where T : class
{
    // everything you already have ...

    return factoryAsync(type, state, checkpoint);
}
```

I am unsure if this makes it any easier lol

Reply:
Yeah I wanted to do that first, then reverted to what I had now